conn_test.go

```diff
@@ -299,15 +308,66 @@ func (s *testRetryPolicy) AttemptTimeout() time.Duration {
 	return s.attemptTimeout
 }
 
-type testQueryObserver struct{}
+type testQueryObserver struct {
+	metrics map[string]*queryMetrics
+	verbose bool
+}
 
 func (o *testQueryObserver) ObserveQuery(ctx context.Context, q ObservedQuery) {
-	Logger.Printf("Observed query %q. Returned %v rows, took %v on host %q. Error: %q\n", q.Statement, q.Rows, q.End.Sub(q.Start), q.Host.ConnectAddress().String(), q.Err)
+	if o.metrics == nil {
+		o.metrics = make(map[string]*queryMetrics)
+	}
+	host := q.Host.ConnectAddress().String()
+	o.metrics[host] = q.Metrics
+	if o.verbose {
+		Logger.Printf("Observed query %q. Returned %v rows, took %v on host %q. Error: %q\n", q.Statement, q.Rows, q.End.Sub(q.Start), host, q.Err)
+	}
+
+	// Uncomment and add verbosity?
```

Comment:
Drop and move the missing log entries `q.Metrics.attempts` and `q.Metrics.totalLatency` into the log message above.